compiler/rustc_ast/src/ast.rs

```diff
@@ -2788,6 +2788,8 @@ pub enum VariantData {
     /// Struct variant.
     ///
     /// E.g., `Bar { .. }` as in `enum Foo { Bar { .. } }`.
+    ///
+    /// The `bool` is whether it was recovered by the parser.
     Struct(ThinVec<FieldDef>, bool),
```

Comment:
```suggestion
    Struct { fields: ThinVec<FieldDef>, recovered: bool },
```

Personally speaking, I'd prefer the variant fields to be named. Not sure how big the fallout of this change would be though.

Reply:
```
gh-aDotInTheVoid@dev-desktop-eu-1:~/rust$ rg "VariantData::Struct" | wc -l
52
```

Thats doable, if people are OK with the churn. Not sure what exactly the policy is, but I'd be happy to do that if it'd be accepted.

Reply:
Yes, go ahead and do that. Please just make sure it's in a separate commit.

Reply:
I'm also wondering if that field should be `Option<ErrorGuaranteed>` instead of `bool`. Can be a separate PR, but maybe leave a FIXME?

```
// FIXME: investigate making this a `Option<ErrorGuaranteed>`
```

Reply:
> Please just make sure it's in a separate commit.

Seperate to what? I think the best thing to do would be do open a new PR, as it wouldn't have any of the changes from this diff.

Reply:
Separate commit as in separate commit in this PR, lol. I don't see any need to split it out from this PR, but up to you.